tests/suite-unit/PluginFlyvemdmPolicyRemoveapplication.php

```diff
@@ -0,0 +1,279 @@
+      if (true == $expected['return']) {
+         // the fleet must exist if we expect a true action.
+         $mockInstance = $this->createFleetInDB();
+      }
+      $result = $policy->apply($mockInstance, $data[0], $data[1], $data[2]);
+      $this->boolean($result)->isEqualTo($expected['return']);
+      if (!$expected['return'] && isset($expected['message'])) {
+         $this->string($_SESSION["MESSAGE_AFTER_REDIRECT"][1][0])->isEqualTo($expected['message']);
+      }
+      unset($_SESSION["MESSAGE_AFTER_REDIRECT"]); // to clear the buffer
+   }
+
+   /**
+    * @tags testGetMqttMessage
+    */
+   public function testGetMqttMessage() {
+      list($policy) = $this->createNewPolicyInstance();
+      $this->boolean($policy->getMqttMessage(null, null, null))->isFalse();
+
+      $mockInstance = $this->newMockInstance('\PluginFlyvemdmPackage');
+      $mockInstance->getMockController()->getID = '1';
+      $mockInstance->getMockController()->getFromDB = true;
+      $mockInstance->getMockController()->getField = 'myFakePackage';
+
+      $result = $policy->getMqttMessage('', $mockInstance, $mockInstance->getID());
+      $this->array($result)->hasKeys(['id', $this->dataField['symbol']])
+         ->string($result['id'])->isEqualTo($mockInstance->getID())
+         ->string($result[$this->dataField['symbol']])->isEqualTo($mockInstance->getField('package_name'));
+   }
+
+   /**
+    * @tags testShowValue
+    */
+   public function testShowValue() {
+      list($policy) = $this->createNewPolicyInstance();
+
+      // Invalid package ID
+      $mockInstance = $this->newMockInstance('\PluginFlyvemdmTask');
+      $mockInstance->getMockController()->getField = -1;
+      $result = $policy->showValue($mockInstance);
+      $this->string($result)->isEqualTo(NOT_AVAILABLE);
+   }
+
+   /**
+    * Create an application (directly in DB) because we are not uploading any file
+    * @return \PluginFlyvemdmPackage
+    */
+   private function createAppInDB() {
+      global $DB;
+
+      $uniqid = uniqid();
+      $table_file = \PluginFlyvemdmPackage::getTable();
+      $query = "INSERT INTO `$table_file` (
+        `package_name`,
+        `alias`,
+        `version`,
+        `filename`,
+        `entities_id`,
+        `dl_filename`,
+        `icon`
+        ) VALUES (
+        '" . $uniqid . $this->packageName . "',
+        'application',
+        '1.0.5',
+        '0/" . $uniqid . $this->filename . "',
+        '0',
+        '" . $this->filename . "',
+        ''
+        )";
+      $result = $DB->query($query);
+      $this->boolean($result)->isTrue();
+
+      $file = new \PluginFlyvemdmPackage();
+      $file->getFromDB($DB->insert_id());
+      $this->boolean($file->isNewItem())->isFalse();
+
+      return $file;
+   }
+
+   /**
+    * Create an application (directly in DB) because we are not uploading any file
+    * @return \PluginFlyvemdmPackage
+    */
+   private function createFleetInDB() {
+      global $DB;
```

Comment:
I think there is no need to create a fleet with a direct SQL query. I think other unit tests create fleets the regular way. If I'm right, let's mutualize this method in TestingCommonTools, and in a code refactor PR (see previous comment), let's mutualize it.